src/m365/spo/commands/spo-search.ts

```diff
@@ -233,23 +233,19 @@ class SpoSearchCommand extends SpoCommand {
         responseType: 'json'
       };
 
-      return request.get(requestOptions);
-    })()
-      .then((searchResult: SearchResult): SearchResult => {
-        resultSet.push(searchResult);
-
-        return searchResult;
-      })
-      .then((searchResult: SearchResult): Promise<SearchResult[]> => {
-        if (args.options.allResults) {
-          if (startRow + searchResult.PrimaryQueryResult.RelevantResults.RowCount < searchResult.PrimaryQueryResult.RelevantResults.TotalRows) {
-            const nextStartRow = startRow + searchResult.PrimaryQueryResult.RelevantResults.RowCount;
-            return this.executeSearchQuery(logger, args, webUrl, resultSet, nextStartRow);
-          }
-        }
-        return new Promise<SearchResult[]>((resolve) => { resolve(resultSet); });
-      })
-      .then(() => resultSet);
+      const searchResult: SearchResult = await request.get(requestOptions);
+      resultSet.push(searchResult);
+
+      if (args.options.allResults && startRow + searchResult.PrimaryQueryResult.RelevantResults.RowCount < searchResult.PrimaryQueryResult.RelevantResults.TotalRows) {
+        const nextStartRow = startRow + searchResult.PrimaryQueryResult.RelevantResults.RowCount;
+        return this.executeSearchQuery(logger, args, webUrl, resultSet, nextStartRow);
+      }
+
+      return resultSet;
+    }
+    catch (err) {
```

Comment:
If you are throwing the error here we might as well remove this entire try catch statement.